api/src/wfl/module/covid.clj

```diff
@@ -271,8 +298,36 @@
         (update :fromSource edn/read-string))
     (throw (ex-info "Invalid executor_items" details))))
 
+(defn ^:private peek-terra-executor-queue
```

Comment:
Can your peek implementation return the workflow instead of the workflow id? Add a line to fetch the workflow from firecloud? This is because the executor is viewed as "a queue of succeeded workflows" by the sink.

Reply:
To be clear, your sink implementation expects and would use the full workflow?  Because we call the `peek` within the `pop`, this change would result in two external calls with one of them unused.  Any appetite instead for calling the fetch from within your update?

For the analogous `peek-tdr-source-queue` implementation, my executor update loop would not make use of the full submission at any point, so I would leave that as is.

Reply:
> To be clear, your sink implementation expects and would use the full workflow? 
Yes

> Because we call the peek within the pop, this change would result in two external calls with one of them unused.
Then implement pop without peek?

> Any appetite instead for calling the fetch from within your update?
No.

Reply:
Okay.

> What do submissions have to do with the implementation of the source queue?

I mistyped and meant snapshot, not submission, corrected above.  I know that you and Rex spoke yesterday about adding the full snapshot either to the source details table or the `peek-tdr-source-queue` implementation.  I am stating explicitly here that I do not need the full snapshot, because this situation is similar and I want to call out the difference in approach as a choice and not an oversight.

Reply:
I'm aware - you just need the ID and the name.

Reply:
We can either store the snapshot in the database or call the datarepo on peek.

Reply:
Fulfilled your request and broke up the peek code so that we don't trigger an unneeded external call when popping, resolving this convo but LMK if you need something else here.